src/visiomode/models/Session.py

```diff
@@ -0,0 +1,57 @@
+#  This file is part of visiomode.
+#  Copyright (c) 2020 Constantinos Eleftheriou <Constantinos.Eleftheriou@ed.ac.uk>
+#  Copyright (c) 2024 Olivier Delree <odelree@ed.ac.uk>
+#  Distributed under the terms of the MIT Licence.
+
+import datetime
+import socket
+import typing
+
+import pydantic
+
+import visiomode.__about__ as __about__
+import visiomode.database as database
+import visiomode.protocols as protocols
+import visiomode._models.Animal as Animal
+import visiomode._models.Trial as Trial
+
+
+class Session(pydantic.BaseModel, arbitrary_types_allowed=True):
+    session_id: str = pydantic.Field(default=None, repr=True)
+    animal_id: str = pydantic.Field(repr=False)
+    duration: pydantic.FiniteFloat = pydantic.Field(repr=False)
+    experiment: str = pydantic.Field(repr=False)
+    protocol: protocols.Protocol = pydantic.Field(repr=False)
+    animal_metadata: typing.Optional[dict] = pydantic.Field(default={}, repr=False)
+    device: str = pydantic.Field(default_factory=socket.gethostname, repr=False)
+    complete: pydantic.StrictBool = pydantic.Field(default=False, repr=False)
+    notes: str = pydantic.Field(default="", repr=False)
+    spec: typing.Optional[dict] = pydantic.Field(default=None, repr=False)
+    timestamp: pydantic.NaiveDatetime = pydantic.Field(
+        default_factory=datetime.datetime.now, repr=True
+    )
+    trials: list[Trial.Trial] = pydantic.Field(default=[], repr=False)
+    version: str = pydantic.Field(default=__about__.__version__, repr=False)
+
+    def model_post_init(self, __context: typing.Any) -> None:
+        self.session_id = self.generate_session_id()
+        # Don't really know why but somehow by this point, Animal has turned from the
+        # module into the class (at runtime, according to Pydantic)? Even Pycharm is
+        # confused.
```

Comment:
I think this might be resolved by addressing the above comments -- it's one of the reasons im not such a big fan of module names mirroring the class names inside them.